robolectric-resources/src/main/java/org/robolectric/res/PackageResourceIndex.java

```diff
@@ -0,0 +1,74 @@
+package org.robolectric.res;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+public class PackageResourceIndex implements ResourceIndex {
+  private static final Logger LOGGER = Logger.getLogger(ResourceExtractor.class.getName());
+
+  private final BiMap<Integer, ResName> resourceTable = HashBiMap.create();
+
+  private Integer maxUsedInt = null;
+  private Integer generatedIdStart = null;
+  private String packageName;
+  private int packageIdentifier;
+
+  public PackageResourceIndex(String packageName) {
+    this.packageName = packageName;
+  }
+
+  @Override
+  public synchronized Integer getResourceId(ResName resName) {
+    Integer id = resourceTable.inverse().get(resName);
+    if (id == null && ("android".equals(resName.packageName) || "".equals(resName.packageName))) {
```

Comment:
I extracted it to a method inside of the same class because this whole logic needs to move. The reason it is here is that Android's own R classes are generated in a different way (through a script I think) that separates the android.R and com.android.internal.R and is also not fully complete (like application generated R values from aapt). So we should actually do this generation as we're loading resources into the ResourceTable (although I think this could still leave us with "holes" as +@id values we wouldn't notice unless we parse the Android XML layout files. Generating the framework arsc files and including them in the platform jar and parsing those directly might be a good way of avoiding all these things entirely.